### ISSUE TITLE
NSFS | S3 | Versioning: Object and delete marker versionId lost after switching to version-suspended mode and object is overwritten

### ISSUE THREAD

**Issue by hseipp** [noobaa/noobaa-core#8381]
### Environment info
- NooBaa Version: **VERSION**
- Platform: **Kubernetes 1.14.1 | minikube 1.1.1 | OpenShift 4.1 | other: specify**

### Actual behavior

Deleted object VersionId and DeleteMarker VersionId are set to null after swtiching to version-suspended mode and object got overwritten.

```
>       assert versions[1]['VersionId'] == version_id_1
E       AssertionError: assert 'null' == 'mtime-d49b8l4g9f5s-ino-k5c'
```

### Expected behavior

Object VersionId and DeleteMarker VersionId should be preserved.

### Steps to reproduce

Execute the following test code as addition to Ceph s3-tests:

```
def test_s3cever103():
    client = get_client()
    bucket_name = get_new_bucket(client=client)

    # The following operation uses another client instance -> following operations will fail
#    check_configure_versioning_retry(bucket_name, "Enabled", "Enabled")
    _ces_safe_check_configure_versioning_retry(client, bucket_name, "Enabled", "Enabled")

    key_name = 'obj'
    data = 'object-1'
    response = client.put_object(Bucket=bucket_name, Key=key_name, Body=data)
    version_id_1 = None
    try:
        version_id_1 = response['VersionId']
    except KeyError:
        version_id_1 = None
    assert version_id_1 != None

    client.delete_object(Bucket=bucket_name, Key=key_name)

    response = client.list_object_versions(Bucket=bucket_name)
    versions = response['Versions']
    assert len(versions) == 1
    assert versions[0]['VersionId'] == version_id_1
    delete_markers = response['DeleteMarkers']
    assert len(delete_markers) == 1
    assert delete_markers[0]['VersionId'] != 'null'

    _ces_safe_check_configure_versioning_retry(client, bucket_name, "Suspended", "Suspended")

    key_name = 'obj'
    data = 'object-1-new'
    response = client.put_object(Bucket=bucket_name, Key=key_name, Body=data)
    version_id_2 = None
    try:
        version_id_2 = response['VersionId']
    except KeyError:
        version_id_2 = None
    assert version_id_2 == None

    response = client.list_object_versions(Bucket=bucket_name)
    versions = response['Versions']
    assert len(versions) == 2
    assert versions[0]['VersionId'] == "null"
    assert versions[1]['VersionId'] == version_id_1
    delete_markers = response['DeleteMarkers']
    assert len(delete_markers) == 1
    assert delete_markers[0]['VersionId'] != 'null'
```

### More information - Screenshots / Logs / Other output

Noobaa "all" level log attached
[noobaa_20240918_2.log.gz](https://github.com/user-attachments/files/17041998/noobaa_20240918_2.log.gz)

**Comment by nadavMiz** [comment]
@hseipp this was fixed by #8298, can you verify this fix is included in the version you are testing?

**Comment by nadavMiz** [comment]
closing the issue, as test passes on GPFS machine with the latest changes:
```
[root@tmtscalets-master s3-tests]# S3TEST_CONF=s3tests.conf tox -- ./s3tests_boto3/functional/test_s3.py::test_s3cever103
.pkg: _optional_hooks> python /usr/local/lib/python3.9/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_sdist> python /usr/local/lib/python3.9/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_wheel> python /usr/local/lib/python3.9/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: prepare_metadata_for_build_wheel> python /usr/local/lib/python3.9/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: build_sdist> python /usr/local/lib/python3.9/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
py: install_package> python -I -m pip install --force-reinstall --no-deps /root/s3-tests/.tox/.tmp/package/17/s3tests-0.0.1.tar.gz
py: commands[0]> pytest ./s3tests_boto3/functional/test_s3.py::test_s3cever103
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.9.18, pytest-8.3.3, pluggy-1.5.0
cachedir: .tox/py/.pytest_cache
rootdir: /root/s3-tests
configfile: pytest.ini
collected 1 item                                                                                                                                                                                           

s3tests_boto3/functional/test_s3.py .                                                                                                                                                                [100%]

============================================================================================ 1 passed in 1.39s =============================================================================================
  py: OK (5.64=setup[3.75]+cmd[1.89] seconds)
  congratulations :) (5.76 seconds)
```
there should be a version with the latest changes coming out on sunday. please re-open if issue persists
